src/components/text-avatar/TextAvatar.tsx

```diff
@@ -1,14 +1,25 @@
 import React from 'react';
 
 import { theme } from 'src/styles/constants/theme';
-import styled from 'styled-components';
+import styled, { CSSProperties } from 'styled-components';
 
-type GraidentSquareType = {
+type GradientSquareType = {
   gradientStart: string;
   gradientEnd: string;
 };
 
-const GradientSquare = styled.div<GraidentSquareType>`
+// Used in store list, and when too many classes are generated styled-components complains. This is their recommended solution to make the changes styles inline, rather than generating a CSS class for each one.
+const GradientSquare = styled.div.attrs<GradientSquareType>(
+  (props): { style: CSSProperties } => ({
+    style: {
+      background: `linear-gradient(
+      137deg,
+      ${props.gradientStart},
+      ${props.gradientEnd}
+    )`,
+    },
+  })
+)<GradientSquareType>`
```

Comment:
Do you have any purpose behind just putting a portion of the style and not the part in the back tick here?

Reply:
That's how styled-components said to do it 🤷 . Only inline the parts that change and keep the rest outside to maintain the normal CSS syntax. Inside the `attrs` it is JS CSS syntax (camelCase).

Reply:
Interesting 😆